models/event.py

```diff
@@ -24,29 +29,106 @@
 
 
 class FlowModel(models.Model):
-  id = models.ForeignKey('auth.User', primary_key=True)
+  id = models.OneToOneField('auth.User', primary_key=True)
   flow = FlowField()
   class Meta:
     app_label = 'tracker'
 
 
 class CredentialsModel(models.Model):
-  id = models.ForeignKey('auth.User', primary_key=True)
+  id = models.OneToOneField('auth.User', primary_key=True)
   credentials = CredentialsField()
   class Meta:
     app_label = 'tracker'
 
 
+class TimestampValidator(validators.RegexValidator):
+  regex = r'(?:(?:(\d+):)?(?:(\d+):))?(\d+)(?:\.(\d{1,3}))?$'
+  def __call__(self, value):
+    super(TimestampValidator, self).__call__(value)
+    h,m,s,ms = re.match(self.regex, unicode(value)).groups()
+    if h is not None and int(m) >= 60:
+      raise ValidationError('Minutes cannot be 60 or higher if the hour part is specified')
+    if m is not None and int(s) >= 60:
+      raise ValidationError('Seconds cannot be 60 or higher if the minute part is specified')
+
+
+class TimestampField(models.Field):
+  __metaclass__ = models.SubfieldBase
+  default_validators = [TimestampValidator()]
+  match_string = re.compile(r'(?:(?:(\d+):)?(?:(\d+):))?(\d+)(?:\.(\d+))?')
+
+  def __init__(self, always_show_h=False, always_show_m=False, always_show_ms=False, *args, **kwargs):
+    super(TimestampField, self).__init__(*args, **kwargs)
+    self.always_show_h = always_show_h
+    self.always_show_m = always_show_m
+    self.always_show_ms = always_show_ms
+  def to_python(self, value):
```

Comment:
I'm so very confused by this method. So, if `value` is an `int`, it converts it to a `string`, if value is a `string`, it converts it to an `int, but when I think of what this method should be returning, my mind keeps going back to datetime.timedelta.